### PR TITLE
added containerization and made adjustments to git.py to support running in container

### PR DESCRIPTION
**Pull request by WLIU0628**
Added Dockerfile and requirements.txt. The current requirements may have excess libraries and can probably be pruned in the future to limit the image size. Also, as of now, the Dockerfile copies all of src, including gerrit_git_dir if the program was recently run. Please delete gerrit_git_dir and index_files (as described in the Dockerfile) to limit the size of the docker image.

### CODE REVIEW

File: Dockerfile
Comment (WLIU0628) [inline Dockerfile]: should I edit the README.md to add this?

Comment (dlatypov) [reply Dockerfile]: Yes, I'd include this in a section in the README.md as well.

I'd imagine most people would assume initially treat the Dockerfile as a blackbox and not look at it initially.

File: src/git.py
Comment (dlatypov) [inline src/git.py]: I assume we can drop this main() now?

I don't think we're expecting people to have issues cloning the repo.

File: .gitignore
Comment (dlatypov) [inline .gitignore:12]: nit: I'd group the src/ dirs together

File: .gitignore
Comment (dlatypov) [inline .gitignore]: potentially add a comment above

e.g.
```
# These directories are generated at runtime <maybe elaborate more>
```

**Review by dlatypov — COMMENTED** [review]
commit description:
Perhaps we can shorten the subject (the initial line) a bit by changing "added containerization" to "Added Dockerfile"

Also, might be worth calling out the issue highlighted by the TODO.
We currently don't have a good identity to use for the committer when running.

Could also make it a bit more readable with some whitespace, e.g.
```
Added Dockerfile and requirements.txt.
The current requirements may have some extra libraries and could be trimmed in the future... .

Note: the Dockerfile copies all of src/, including generated dirs like gerrit_git_dir/.
Please delete gerrit_git_dir to limit the size of the docker image.
```

File: README.md
Comment (dlatypov) [inline README.md]: nit:drop space between ` and bash

File: README.md
Comment (dlatypov) [inline README.md]: nit: use `` around the names.

Then you won't have to escape the _

File: src/git.py
Comment (dlatypov) [inline src/git.py]: we don't need the local variable here
```
try:
   return self._git.push(...)
```

File: README.md
Comment (dlatypov) [inline README.md]: nit: "###" to match the others

File: README.md
Comment (dlatypov) [inline README.md]: optional: can intead give a command to copy paste like

```bash
$ rm -rf linux_kselftest/ src/gerrit_git_dir/ src/index_files/
```

**Review by dlatypov — COMMENTED** [review]
Overall LGTM, just minor nits here and there.

**Review by dlatypov — APPROVED** [review]
LGTM.
I'll hold off on merging for now in case Brendan wants to take a look.

**Review by bjh83 — APPROVED** [review]
LGTM.

**Pull request merged**
